src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedRootModuleTypeStaticConstructor.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRootModuleTypeStaticConstructor : MethodSymbol
+    {
+        private readonly RootModuleType _rootModuleType;
+
+        internal SynthesizedRootModuleTypeStaticConstructor(RootModuleType rootModuleType, NamedTypeSymbol containingType)
+        {
+            _rootModuleType = rootModuleType;
+            ContainingType = containingType;
+        }
+
+        public override NamedTypeSymbol ContainingType { get; }
+
+        public override Symbol ContainingSymbol => ContainingType;
+
+        public override string Name => WellKnownMemberNames.StaticConstructorName;
+
+        internal override bool HasSpecialName => true;
+
+        public override bool HidesBaseMethodsByName => true;
+
+        internal override bool SynthesizesLoweredBoundBody => true;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            F.CurrentFunction = this;
+
+            var body = ArrayBuilder<BoundStatement>.GetInstance();
+
+            foreach (var moduleInitializerMethod in _rootModuleType.GetModuleInitializerMethods())
+            {
+                body.Add(F.ExpressionStatement(
+                    F.StaticCall((MethodSymbol)moduleInitializerMethod, ImmutableArray<BoundExpression>.Empty)));
```

Comment:
This MethodSymbol cast bothered me a bit (should there be a CSharpRootModuleType that holds MethodSymbols instead of Cci.IMethodDefinitions?) but then I saw that SynthesizedPrivateImplementationDetailsStaticConstructor.GenerateMethodBody does the exact same kind of cast:

https://github.com/dotnet/roslyn/blob/db5c01e59a0fea97281a1adde921c097f48e73e6/src/Compilers/CSharp/Portable/Emitter/Model/SynthesizedPrivateImplementationDetailsStaticConstructor.cs#L48